### PR TITLE
Added functions to init file

### PR DESCRIPTION
**Pull request by obriente**
Finally fixed my commit history!

I've gone through and either exposed or made explicitly hidden all (I think) functions in the repo. Some of the names were a bit vague (e.g. 'two_body'), LMK if you think something should be exposed that isn't or vice versa.

### CODE REVIEW

**Comment by ncrubin** [comment]
@obriente Even resolving the merge conflict it still seems like there is an issue with importing.  Maybe cutting this down to something less than 21 files.  Maybe just bumping this back to when it was passing tests.  Then we can add the extra files in a following PR.

**Comment by ncrubin** [comment]
@googlebot I fixed it.

**Comment by ncrubin** [comment]
@googlebot I signed it.

**Comment by ncrubin** [comment]
@googlebot I consent.

**Comment by ncrubin** [comment]
@obriente bump. I went in and cleaned up some of the imports that were broken from the move of a method.  Let me know if everything else looks in place.

File: src/openfermion/transforms/opconversions/commutator_diagonal_coulomb_operator.py [hunk 1]
```diff
@@ -14,8 +14,9 @@
 import warnings
 
 from openfermion.ops.operators import FermionOperator
-import openfermion.transforms.opconversions.term_reordering\
-    as term_reordering
+from openfermion.transforms.opconversions import term_reordering
+# import openfermion.transforms.opconversions.term_reordering\
+#     as term_reordering
```
Comment (obriente) [inline src/openfermion/transforms/opconversions/commutator_diagonal_coulomb_operator.py:19]: Not sure why this is commented out, maybe we can just delete it? (No idea if this was from me or you @ncrubin ).

**Comment by obriente** [comment]
Thanks for sorting that out, looks like it must have been a pain to fix.

Everything looks good to me, I guess we just need one extra review now.

File: src/openfermion/transforms/opconversions/commutator_diagonal_coulomb_operator.py (hunk 1 above)
Comment (ncrubin) [reply src/openfermion/transforms/opconversions/commutator_diagonal_coulomb_operator.py:19]: That was me.  I was sorting out what was causing the issue and I changed this to a from import.  I'll fix it.

**Comment by ncrubin** [comment]
No need for extra review.  thanks for kicking this off and sorting out the extra imports that users were not finding.  This is an impactful PR.

File: src/openfermion/transforms/opconversions/commutator_diagonal_coulomb_operator.py (hunk 1 above)
Comment (obriente) [reply src/openfermion/transforms/opconversions/commutator_diagonal_coulomb_operator.py:19]: Ah ok. Did that make a difference? I feel a bit lost in this damn mess of imports we have.

**Comment by obriente** [comment]
Ah ok - I can't merge right now without an extra review, maybe you have more privileges than I do here. But everything LGTM

File: src/openfermion/transforms/opconversions/commutator_diagonal_coulomb_operator.py (hunk 1 above)
Comment (ncrubin) [reply src/openfermion/transforms/opconversions/commutator_diagonal_coulomb_operator.py:19]: The problem wasn't the importing structure.  A file that got moved `double_commuting...two_body` was being incorrectly imported (wrong location).  Thus tests were failing.  The error message was different from when you get circular imports...though sounded similar.  

All that I did to fix was to import from the correct location.

Comment (obriente) [reply src/openfermion/transforms/opconversions/commutator_diagonal_coulomb_operator.py:19]: Hah, very sorry about that then. For some reason tests were passing on my computer... Or at least I thought they were. (I mean, this isn't the kind of error that should pass on any machine)

**Pull request merged**